cli/data/enti/download.py

```diff
@@ -4,18 +4,77 @@
 #
 # Copyright (C) 2022 Giacomo Tesio <giacomo@tesio.it>
 # Copyright (C) 2022 Leonardo Canello <leonardocanello@protonmail.com>
+# Copyright (C) 2022 Andrea Foletto <andrea@yaaaw.it>
 #
 # MonitoraPA is a hack. You can use it according to the terms and
 # conditions of the Hacking License (see LICENSE.txt)
 
 import sys
 import requests
-import commons
-
-outDir = commons.computeOutDir(sys.argv)
-url = 'https://indicepa.gov.it/ipa-dati/datastore/dump/d09adf99-dc10-4349-8c53-27b1e5aa97b6?bom=True&format=tsv'
-r = requests.get(url, allow_redirects=True)
-result = r.content.replace(
-    b'HTTPS:SISTEMAAMBIENTELUCCA.IT', b'HTTPS://SISTEMAAMBIENTELUCCA.IT')
-open("%s/enti.tsv" % outDir, 'wb').write(result)
-print("%s/enti.tsv" % outDir)
+import datetime
+import os
+import shutil
+
+def verifyExecutionDirectory():
+    if not os.path.isdir("cli") or not os.path.isfile("LICENSE.txt"):
+        print("[ Error ] Please run cli/ scripts from the root directory")
+        sys.exit(1)
+
+    if not os.path.isdir("out"):
+        os.mkdir("out")
+
+
+def computeOutDir(argv):
+
+    verifyExecutionDirectory()
+
+    dirName = f"out/{datetime.datetime.utcnow().strftime('%Y-%m-%d')}"
+
+    if not os.path.isdir(dirName):
+        os.mkdir(dirName)
```

Comment:
forse intendevi os.makedirs(dirname)?

secondo quello che ci dicevamo in chat dovrebbe essere qualcosa tipo
```suggestion
        try:
            os.makedirs(dirName)
        except OSError as exc: 
            if exc.errno == errno.EEXIST and os.path.isdir(dirName):
                pass
```

Reply:
Io preferirei un approccio senza `try-except`, al massimo possiamo aggiungere il controllo che nella directory possiamo scrivere.

Reply:
ok, prova: vediamo quale soluzione viene più leggibile